classes/core/AltapaySettings.php

```diff
@@ -1,392 +1,409 @@
+
+	/**
+	 * AltapaySettings constructor.
+	 */
+	public function __construct() {
+		 // Load localization files
+		add_action( 'init', array( $this, 'altapayLocalizationInit' ) );
+		// Add admin menu
+		add_action( 'admin_menu', array( $this, 'altapaySettingsMenu' ), 60 );
+		// Register settings
+		add_action( 'admin_init', array( $this, 'altapayRegisterSettings' ) );
+		// Add settings link on plugin page
+		add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), array( $this, 'addActionLinks' ) );
+		// Order completed interceptor:
+		add_action( 'woocommerce_order_status_completed', array( $this, 'altapayOrderStatusCompleted' ) );
+
+		add_action( 'admin_notices', array( $this, 'loginError' ) );
+		add_action( 'admin_notices', array( $this, 'captureFailed' ) );
+		add_action( 'admin_notices', array( $this, 'captureWarning' ) );
+	}
+
+	/**
+	 * @param int $orderID
+	 * @return void
+	 * @throws AltapayMerchantAPIException
+	 */
+	public function altapayOrderStatusCompleted( $orderID ) {
+		$this->startSession();
+		// Load order
+		$order = new WC_Order( $orderID );
+		$txnID = $order->get_transaction_id();
+
+		if ( ! $txnID ) {
+			return;
+		}
+
+		$api = $this->apiLogin();
+		if ( $api instanceof WP_Error ) {
+			$_SESSION['altapay_login_error'] = $api->get_error_message();
+			echo '<p><b>' . __( 'Could not connect to AltaPay!', 'altapay' ) . '</b></p>';
+			return;
+		}
+
+		$payment = $api->getPayment( $txnID );
+		if ( ! $payment ) {
+			return;
+		}
+
+		$payments = $payment->getPayments();
+		$pay      = $payments[0];
+
+		if ( $pay->getCapturedAmount() > 0 ) {
+			$this->saveCaptureWarning( 'This order was already fully or partially captured: ' . $orderID );
+		} else { // Order wasn't captured and must be captured now.
+			$amount        = $pay->getReservedAmount(); // Amount to capture.
+			$salesTax      = $order->get_total_tax();
+			$orderLines    = array( array() );
+			$captureResult = $api->captureReservation( $txnID, $amount, $orderLines, $salesTax );
+
+			if ( ! $captureResult->wasSuccessful() ) {
+				$order->add_order_note(
+					__(
+						'Capture failed: ' . $captureResult->getMerchantErrorMessage(),
+						'Altapay'
+					)
+				); // log to history.
+				$this->saveCaptureFailedMessage(
+					'Capture failed for order ' . $orderID . ': ' . $captureResult->getMerchantErrorMessage()
+				);
+
+				return;
+			} else {
+				update_post_meta( $orderID, '_captured', true );
+				$order->add_order_note( __( 'Order captured: amount: ' . $amount, 'Altapay' ) );
+			}
+		}
+	}
+
+
+	/**
+	* Starts the session
+	*
+	* @return void
+	*/
+	public function startSession() {
+		if ( session_id() === '' ) {
+			session_start();
+		}
+	}
+
+	/**
+	 * @param string $newMessage
+	 * @return void
+	 */
+	public function saveCaptureFailedMessage( $newMessage ) {
+		if ( isset( $_SESSION['altapay_capture_failed'] ) ) {
+			$message = $_SESSION['altapay_capture_failed'] . '<br/>';
+		} else {
+			$message = '';
+		}
```

Comment:
Set empty string as the default instead of using else.
```suggestion
		$message = '';
		if ( isset( $_SESSION['altapay_capture_failed'] ) ) {
			$message = $_SESSION['altapay_capture_failed'] . '<br/>';
		}
```